jetty-server/src/main/java/org/eclipse/jetty/server/ResourceService.java

```diff
@@ -377,22 +377,19 @@ protected void sendWelcome(HttpContent content, String pathInContext, boolean en
         if (!endsWithSlash)
         {
             StringBuffer buf = request.getRequestURL();
-            synchronized (buf)
```

Comment:
Don't remove this `synchronized`.   The StringBuffer is synchronized no matter what we do, so the outer synchronized prevents the lock being grabbed and released for every call.